crypto-jni/src/lib.rs

```diff
@@ -665,3 +672,445 @@ pub fn serialize_init_parameters_aux(
         .serial_value(&value)
         .map_err(|e| anyhow!("{}", e.display(verboseErrors)))
 }
+
+fn get_string(env: JNIEnv, java_string: JString) -> Result<String, JNIErrorResponse> {
+    let java_str = match env.get_string(java_string) {
+        Ok(s) => s,
+        Err(err) => {
+            return Err(JNIErrorResponse {
+                errorMessage: err.to_string(),
+                errorType:    JNIErrorResponseType::ParameterSerialization,
```

Comment:
Strictly speaking I believe that this is an instance of a `JsonDeserialization` error.

As the FFI between Java and Rust communicates via Json. 

The `ParameterSerialization` error variant is used for indicating smart contract parameter serialization failure. 

Admittedly, this is not really well documented at the moment.

Reply:
I have changed the error type.